docker-compose.yml

```diff
@@ -12,77 +12,18 @@ services:
       - "sshvolume:/etc/letsencrypt"
     ports:
       - 443:443
+      - 10000:10000
+      - 10001:10001
     deploy:
       replicas: 1
       placement:
         constraints: [node.labels.node_role == devops]
     networks:
       - proxy
+    configs:
+      - clin-external-proxy-v5
     command: |
-      /bin/bash -c "echo '
-      server {
-        listen 443 ssl;
-        add_header X-Frame-Options "SAMEORIGIN";
-        root /usr/share/nginx/html;
-        index index.html;
-        error_page 404 = @notfound;
-
-        # Directive for clin-frontend
-        location @notfound {
-          return 302 /?redirect=$$request_uri;
-        }
-
-        server_name clin.cr-ste-justine.xyz;
-        ssl_certificate /etc/letsencrypt/live/clin.cr-ste-justine.xyz/fullchain.pem;
-        ssl_certificate_key /etc/letsencrypt/live/clin.cr-ste-justine.xyz/privkey.pem;
-
-        # Directives for clin-proxy-api
-        location /api/auth {
-          proxy_pass https://auth:3000/auth;
-          proxy_http_version 1.1;
-          proxy_set_header Connection keep-alive;
-          proxy_set_header Upgrade $$http_upgrade;
-          proxy_set_header Host $$host;
-          proxy_set_header X-Real-IP $$remote_addr;
-          proxy_cache_bypass $$http_upgrade;
-        }
-        location /api/patient {
-          proxy_pass https://patient:4000/patient;
-          proxy_http_version 1.1;
-          proxy_set_header Connection keep-alive;
-          proxy_set_header Upgrade $$http_upgrade;
-          proxy_set_header Host $$host;
-          proxy_set_header X-Real-IP $$remote_addr;
-          proxy_cache_bypass $$http_upgrade;
-        }
-        location /api/variant {
-          proxy_pass https://variant:5001/variant;
-          proxy_http_version 1.1;
-          proxy_set_header Connection keep-alive;
-          proxy_set_header Upgrade $$http_upgrade;
-          proxy_set_header Host $$host;
-          proxy_set_header X-Real-IP $$remote_addr;
-          proxy_cache_bypass $$http_upgrade;
-        }
-        location /api/meta {
-          proxy_pass https://meta:7000/meta;
-          proxy_http_version 1.1;
-          proxy_set_header Connection keep-alive;
-          proxy_set_header Upgrade $$http_upgrade;
-          proxy_set_header Host $$host;
-          proxy_set_header X-Real-IP $$remote_addr;
-          proxy_cache_bypass $$http_upgrade;
-        }
-        location /api/gene {
-          proxy_pass https://gene:5002/gene;
-          proxy_http_version 1.1;
-          proxy_set_header Connection keep-alive;
-          proxy_set_header Upgrade $$http_upgrade;
-          proxy_set_header Host $$host;
-          proxy_set_header X-Real-IP $$remote_addr;
-          proxy_cache_bypass $$http_upgrade;
-        }
-      }' | tee -i /etc/nginx/conf.d/default.conf && nginx -g 'daemon off;'"
+      /bin/bash -c "cp /clin-external-proxy-v5 /etc/nginx/conf.d/default.conf && nginx -g 'daemon off;'"
```

Comment:
Here: https://github.com/cr-ste-justine/clin-external-proxy

I'm not completely happy with it. Its an initial 50% decoupling effort, but the real way of doing it would be to have separate images (one dedicated to serving the frontend and one dedicated to being an external-facing proxy for everything in clin... the beginnings of a gateway service if you will).

I'll do this properly before the weekend.